### ISSUE TITLE
Callback* not triggered 

### ISSUE THREAD

**Issue by 3wer3wer** [rekab-app/background_locator#301]
After the update from 1.6.6 to 1.6.12 I am facing the problem that the Callback-print* (see below) is not triggered even if the plugin is running. 

I receive the message: D/plugin (4622): sendLocationEvent... but the listener is not triggered to update the UI. I tested with 2 devices and 2 emulators. Maybe I forgot to update something in the AppDelegate.swift and build.gradle. 

I am pretty new to flutter. Sorry if I forget something obvious!

Thank you in advance and have a nice week.

*
 Future<void> callback(LocationDto locationDto) async {
    print('$_count location in dart: ${locationDto.toString()}');
    await setLogPosition(_count, locationDto);
    final SendPort? send = IsolateNameServer.lookupPortByName(isolateName);
    send?.send(locationDto);
    _count++;
  }

**Comment by nick1377** [comment]
Same. Looks like the code is throwing a Null is not a String error because the provider in the location json is missing.

**Comment by rulefahd** [comment]
true++++++++++

**Comment by AndroidDev68** [comment]
same problem

**Comment by n0tc0ding** [comment]
same problem

**Comment by justjcool** [comment]
Same problem

**Comment by frankvollebregt** [comment]
> Same. Looks like the code is throwing a Null is not a String error because the provider in the location json is missing.

I made a fork where I set it to an empty String by default, which seems to indeed solve the issue.
```yaml
dependency_overrides:
  background_locator:
    git:
      url: git@github.com:frankvollebregt/background_locator.git
      ref: location-dto-null-fix
```
It should probably be included in the actual JSON instead, but for now this might help

**Comment by CristianKoh1** [comment]
I solved it in the following way
![image](https://user-images.githubusercontent.com/45083832/150855439-ece79785-c213-43da-b7d2-0f0aa2f70634.png)

**Comment by frankvollebregt** [comment]
Similar to what I did, I used a slightly shorter notation to do the same thing
```dart
...
json[Keys.ARG_PROVIDER] ?? '',
```

**Comment by santa112358** [comment]
@SaeedMasoumi @easazade @payam-zahedi 
This package does not work on latest version. We all need to use the forked branch at the moment.
Please solve this issue as soon as possible, as it seems critical and will not take much time to fix.

Thank you.

**Comment by payam-zahedi** [comment]
@santa112358 
It seems that this package has not maintained for a long time.
I will work on this package this weekend.

**Comment by payam-zahedi** [comment]
@santa112358 @frankvollebregt @CristianKoh1 
seems we have open PR that worked on this issue. 
#317 
I think this will solve the problem.

**Comment by frankvollebregt** [comment]
Yeah, there are a few PRs open that could resolve this. It's up to you to pick which one you like best I guess. While you're taking a look, fixing the warnings caused by Flutter 3.0 is probably a good idea too :D

Thanks for putting in the time to fix it

**Comment by Usama-dev575** [comment]
It's not working for iOS. Anyone facing the same issue?
